### PR TITLE
Get string value assert elimination

### PR DESCRIPTION
**Pull request by jirimosinger**
- Assert replaced with if statement in `get_string_array_value(...)`
- Added tests for empty arrays
- Minor refactoring of other tests

### CODE REVIEW

File: parser_library/test/processing/ca_instr_test.cpp [hunk 1]
```diff
@@ -27,341 +27,247 @@ using namespace hlasm_plugin::parser_library::semantics;
 
 TEST(var_subs, gbl_instr_only)
 {
-    std::string input("   gbla var");
```
Comment (slavek-kucera) [inline parser_library/test/processing/ca_instr_test.cpp:30]: I am not sure about this... the language is supposed to be case-insensitive in these situations, so I think we should keep the mixed case in the tests.

File: parser_library/test/processing/ca_instr_test.cpp [hunk 2]
```diff
@@ -27,341 +27,247 @@ using namespace hlasm_plugin::parser_library::semantics;
...
-    std::vector<A_t> subscript2;
-    subscript2.push_back(3);
-    tmp = get_var_sym_value(ctx, it, std::move(subscript2), {}, diags).access_a();
-    EXPECT_EQ(tmp, 4);
-    std::vector<A_t> subscript3;
-    subscript3.push_back(4);
-    tmp = get_var_sym_value(ctx, it, std::move(subscript3), {}, diags).access_a();
-    EXPECT_EQ(tmp, 5);
+    std::unordered_map<size_t, A_t> expected = { { 1, 3 }, { 2, 4 }, { 3, 5 } };
+    EXPECT_THAT(*get_var_vector_map<A_t>(a.hlasm_ctx(), "VAR"), ::testing::ContainerEq(expected));
```
Comment (slavek-kucera) [inline parser_library/test/processing/ca_instr_test.cpp:107]: should work, right?

```suggestion
    EXPECT_EQ(get_var_vector_map<A_t>(a.hlasm_ctx(), "VAR"), expected);
```

Comment (jirimosinger) [reply parser_library/test/processing/ca_instr_test.cpp:107]: It does work. I wanted to uses `EXPECT_THAT` as it gives a nicer output message when something goes wrong. Nevertheless, I can change it to `EXPECT_EQ`

File: parser_library/test/processing/ca_instr_test.cpp [hunk 3]
```diff
@@ -27,341 +27,247 @@ using namespace hlasm_plugin::parser_library::semantics;
...
-    auto it2 = ctx.ids().add("var2");
-    auto it3 = ctx.ids().add("var3");
-
-    EXPECT_TRUE(ctx.get_var_sym(it1));
-    EXPECT_TRUE(ctx.get_var_sym(it2));
-    EXPECT_TRUE(ctx.get_var_sym(it3));
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR1"), false);
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR2"), false);
+    EXPECT_EQ(get_var_value<B_t>(a.hlasm_ctx(), "VAR3"), false);
 }
```
Comment (slavek-kucera) [inline parser_library/test/processing/ca_instr_test.cpp:233]: is there a diagnostic produced?

File: parser_library/test/processing/ca_instr_test.cpp (hunk 2 above)
Comment (slavek-kucera) [reply parser_library/test/processing/ca_instr_test.cpp:107]: My main problem is that the `*` on valueless optional is UB. But comparing it should be perfectly fine.

File: parser_library/test/processing/ca_instr_test.cpp (hunk 3 above)
Comment (jirimosinger) [reply parser_library/test/processing/ca_instr_test.cpp:233]: Nope. Should it be? It seems fine when run on the mainframe

File: parser_library/test/processing/ca_instr_test.cpp (hunk 1 above)
Comment (jirimosinger) [reply parser_library/test/processing/ca_instr_test.cpp:30]: No problem

File: parser_library/test/processing/ca_instr_test.cpp (hunk 3 above)
Comment (slavek-kucera) [reply parser_library/test/processing/ca_instr_test.cpp:231]: I was just not sure... looks fine.

**Pull request merged**